Lib/test/test_mmap.py

```diff
@@ -1058,6 +1060,91 @@ def __exit__(self, exc_type, exc_value, traceback):
                 with self.assertRaisesRegex(ValueError, "mmap closed or invalid"):
                     m.write_byte(X())
 
+    @unittest.skipUnless(os.name == 'nt', 'requires Windows')
+    @unittest.skipUnless(hasattr(mmap.mmap, '_protect'), 'test needs debug build')
+    def test_access_violations(self):
+
+        code = textwrap.dedent("""
+            from test.support.os_helper import TESTFN
+            import faulthandler
+            import mmap
+            import os
+            from contextlib import suppress
+
+            # Prevent logging access violations to stderr.
+            faulthandler.disable()
+
+            if os.path.exists(TESTFN):
+                os.unlink(TESTFN)
+
+            PAGESIZE = mmap.PAGESIZE
+            PAGE_NOACCESS = 0x01
+
+            with open(TESTFN, 'bw+') as f:
+                f.write(b'A'* PAGESIZE)
+                f.flush()
+
+                m = mmap.mmap(f.fileno(), PAGESIZE)
+                m._protect(PAGE_NOACCESS, 0, PAGESIZE)
+                with suppress(OSError):
```

Comment:
Pretty sure `suppress` will silently continue if the exception is not raised. Do we want to require that it be raised?

Adding an `assert False, "m.read(...) did not raise"` after the function that should raise ought to handle it.

If we want to pass silently even when the failure is not triggered, then add a comment here mentioning that no exception or OSError are equally acceptable.

Reply:
it should be triggered yes, previously I used assertRaises, but I simplified it for the subprocess method
I will add some asserts